src/unitycatalog/ai/databricks.py

```diff
@@ -484,8 +488,36 @@ def _execute_uc_function(
                     format="CSV", value=csv_buffer.getvalue(), truncated=truncated
                 )
 
-        # TODO: support serverless
-        raise ValueError("warehouse_id is required for executing UC functions in Databricks")
+        else:
+            _logger.info("Using databricks connect to execute functions with serverless compute.")
+            try:
+                self.set_default_spark_session()
+                sql_command = get_execute_function_sql_command(function_info, parameters)
+                result = self.spark.sql(sqlQuery=sql_command)
+                if is_scalar(function_info):
+                    return FunctionExecutionResult(
+                        format="SCALAR", value=str(result.collect()[0][0])
+                    )
+                else:
+                    row_limit = int(
+                        os.environ.get(
+                            UC_AI_CLIENT_EXECUTION_RESULT_ROW_LIMIT,
+                            DEFAULT_UC_AI_CLIENT_EXECUTION_RESULT_ROW_LIMIT,
+                        )
+                    )
+                    truncated = False
+                    if result.count() > row_limit:
+                        truncated = True
+                    pdf = result.limit(row_limit).toPandas()
+                    csv_buffer = StringIO()
+                    pdf.to_csv(csv_buffer, index=False)
+                    return FunctionExecutionResult(
+                        format="CSV", value=csv_buffer.getvalue(), truncated=truncated
+                    )
```

Comment:
can we minimize the try-catch scope?

Reply:
Emm I moved first two lines out. 
I think `result = self.spark.sql(sqlQuery=sql_command)` needs to be in the try catch, so are `result.collect()` and `result.toPandas()` as that's when we do the computation. But still it's quite long code to wrap 🤔